modules/inputSpinner/InputSpinner.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import {
+  ThemeStyleSheetFactory,
+  StandardProps,
+  WithStylesProps
+} from "@diana-ui/types";
+import { withStyles } from "@diana-ui/base";
+import { ButtonText } from "@diana-ui/typography";
+
+// @ts-ignore
+export interface IProps extends StandardProps<"input"> {
+  min?: number;
+  max?: number;
+  step?: number;
+  value?: number;
+  onChange?: (value: number) => void;
+}
+
+const styleSheet: ThemeStyleSheetFactory = theme => ({
+  container: {
+    boxSizing: "border-box",
+    margin: 0,
+    "@selectors": {
+      "&:hover :not(.disabled).value": {}
+    }
+  },
+  button: {
+    outline: "none",
+    cursor: "pointer",
+    "@selectors": {
+      "&.disabled": {
+        cursor: "initial"
+      }
+    }
+  },
+  buttonMin: {},
+  buttonMax: {},
+  value: {
+    boxSizing: "border-box",
+    alignItems: "center",
+    justifyContent: "center",
+    "@selectors": {
+      "&.disabled": {},
+      "&.focus:not(.disabled)": {}
+    }
+  }
+});
+
+const InputSpinner: React.FC<IProps & WithStylesProps> = ({
+  styles,
+  cx,
+  className,
+  value,
+  onChange,
+  min,
+  max,
+  step,
+  disabled
+}) => {
+  const [_value, setValue] = useState(0);
```

Comment:
can't we just use the value / onChange from the above component instead of having an internally state ?
I could be missing something though xd

Reply:
This was made so that this component can either be controlled or uncontrolled. With this it allows him to change value without the need of the parent to control it